rclpy/src/rclpy/_rclpy.c

```diff
@@ -851,6 +852,99 @@ rclpy_count_subscribers(PyObject * Py_UNUSED(self), PyObject * args)
   return _count_subscribers_publishers(args, "subscribers", rcl_count_subscribers);
 }
 
+typedef rcl_ret_t (* rcl_get_info_by_topic_func_t)(
+  const rcl_node_t * node,
+  rcutils_allocator_t * allocator,
+  const char * topic_name,
+  bool no_mangle,
+  rmw_topic_info_array_t * info_array);
+
+static PyObject *
+_get_info_by_topic(
+  PyObject * args, const char * type,
+  rcl_get_info_by_topic_func_t rcl_get_info_by_topic)
+{
+  PyObject * pynode;
+  const char * topic_name;
+  int no_mangle;
+
+  if (!PyArg_ParseTuple(args, "Osp", &pynode, &topic_name, &no_mangle)) {
+    return NULL;
+  }
+
+  rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, "rcl_node_t");
+  if (!node) {
+    return NULL;
+  }
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rmw_topic_info_array_t info_array = rmw_get_zero_initialized_topic_info_array();
+  rcl_ret_t ret = rcl_get_info_by_topic(node, &allocator, topic_name, no_mangle, &info_array);
```

Comment:
@jaisontj @mm318 I can't seem to find `rcl_get_info_by_topic`, but `rcl_get_publishers_info_by_topic` and  `rcl_get_subscribers_info_by_topic` in https://github.com/ros2/rcl/pull/511.

Reply:
`rcl_get_info_by_topic` is of type `rcl_get_info_by_topic_func_t` passed as a param to this function and typedef'd at [line 855](https://github.com/ros2/rclpy/pull/454/files/d85917f7def9f92eaa4b9b83c524ddb305be1786#diff-80dda03110b5be606b823d6b0558b5c2R855).

Reply:
Oof, I didn't see that one. Disregard.